FogFog-iOS/FogFog-iOS/Presentation/Login/ViewController/MakeNicknameViewController.swift

```diff
@@ -0,0 +1,161 @@
+        }
+        
+        titleLabel.do {
+            $0.text = "포그포그에서 사용할\n닉네임을 입력해주세요"
+            $0.numberOfLines = 2
+            $0.font = .pretendardB(24)
+            $0.textAlignment = .left
+        }
+        
+        confirmButton.do {
+            $0.setTitle("확인", for: .normal)
+            $0.titleLabel?.font = .pretendardB(18)
+            $0.titleLabel?.textColor = .white
+            $0.backgroundColor = .grayBlack
+        }
+        
+        errorImageView.do {
+            $0.image = FogImage.errorIcon
+        }
+        
+        errorLabel.do {
+            $0.text = "이미 존재하는 닉네임 입니다."
+            $0.textColor = .etcRed
+            $0.font = .pretendardM(10)
+        }
+    }
+    
+    override func setLayout() {
+        view.addSubviews([naviView, titleLabel, nicknameTextField, confirmButton, backView, errorImageView, errorLabel])
+        
+        naviView.snp.makeConstraints {
+            $0.top.leading.trailing.equalToSuperview()
+            $0.height.equalTo(92)
+        }
+        
+        titleLabel.snp.makeConstraints {
+            $0.top.equalTo(naviView.snp.bottom).offset(52)
+            $0.leading.equalToSuperview().inset(16)
+        }
+        
+        nicknameTextField.snp.makeConstraints {
+            $0.top.equalTo(titleLabel.snp.bottom).offset(41)
+            $0.leading.trailing.equalToSuperview().inset(16)
+            $0.height.equalTo(54)
+        }
+        
+        confirmButton.snp.makeConstraints {
+            $0.leading.trailing.bottom.equalTo(view.safeAreaLayoutGuide)
+            $0.height.equalTo(54)
+        }
+        
+        backView.snp.makeConstraints {
+            $0.top.equalTo(confirmButton.snp.bottom)
+            $0.leading.trailing.bottom.equalToSuperview()
+        }
+        
+        errorImageView.snp.makeConstraints {
+            $0.top.equalTo(nicknameTextField.snp.bottom).offset(10)
+            $0.leading.equalTo(nicknameTextField.snp.leading)
+            $0.size.equalTo(13)
+        }
+        
+        errorLabel.snp.makeConstraints {
+            $0.leading.equalTo(errorImageView.snp.trailing).offset(5)
+            $0.centerY.equalTo(errorImageView.snp.centerY)
+        }
+    }
+    
+    private func bind() {
+        output.nickname
+            .asDriver(onErrorJustReturn: "")
+            .drive(onNext: { [weak self] nickname in
+                self?.nicknameTextField.text = nickname
+            })
+            .disposed(by: disposeBag)
+        
+        output.isValid
+            .asDriver(onErrorJustReturn: false)
+            .drive(onNext: { [weak self] result in
+                [self?.errorImageView, self?.errorLabel].forEach { $0?.isHidden = result }
+                if result {
+                    self?.nicknameTextField.setBoderColor(color: .fogBlue)
+                } else {
+                    self?.nicknameTextField.setBoderColor(color: .etcRed)
+                }
```

Comment:
p3;

```suggestion
                self?.nicknameTextField.setBoderColor(color: result ? .fogBlue : .etcRed)
```
- 다음과 같이 쓸 수도 있을 것 같습니다. (취향 차이)

@jane1choi 
- 궁금한 점이 있는데 처음 텍스트 필드를 탭했을 때에도 validation 검사를 하나요?
- 처음 시작 했을 때 즉, 포커스 됐을 때는 Color가 어떻게 되는지 궁금합니다.

Reply:
@Taehyeon-Kim 
텍스트 필드를 탭하는 경우에도 "" 이 방출되기 때문에 validation 검사가 되어 blue컬러로 바뀝니다!
리뷰를 보니까 사용자가 값을 입력하는 순간부터 검사가 진행되는게 좀 더 자연스러울 것 같다는 생각이 드는데, 이 부분 디자인이랑 논의해보겠습니당 ! 꼼꼼한 확인 감사합니다 ~